mypy/test/testparse.py

```diff
@@ -7,25 +7,29 @@
 from mypy import defaults
 from mypy.myunit import Suite, AssertionFailure
 from mypy.test.helpers import assert_string_arrays_equal
-from mypy.test.data import parse_test_cases, DataDrivenTestCase
+from mypy.test.data import parse_test_cases, DataDrivenTestCase, DataSuite
 from mypy.test import config
 from mypy.parse import parse
 from mypy.errors import CompileError
 from mypy.options import Options
 
 
-class ParserSuite(Suite):
+class ParserSuite(DataSuite):
     parse_files = ['parse.test',
                    'parse-python2.test']
 
-    def cases(self) -> List[DataDrivenTestCase]:
+    @classmethod
```

Comment:
It's called without an instance (line 515 in data.py, obj is the class itself); it could have been a static method. Better yet, it could not exist at all, since the needed information here is only the file names and 2-3 options.